### ISSUE TITLE
why the shape of X_wide_test different from X_wide_train

### ISSUE THREAD

**Issue by TtCWH** [jrzaurin/pytorch-widedeep#12]
train_set and test_set both use preprocess_wide built by WidePreprocessor(wide_cols=wide_cols, crossed_cols=cross_cols)

**Comment by jrzaurin** [comment]
Hey @TtCWH 

Could you elaborate a bit more? or give me some code I can reproduce?

`WidePreprocessor` simply one-hot encodes the categorical features. So the only way I find for the test set to have different shape than the train set would be if there is a new category in the test set. 

However,  `WidePreprocessor` uses `Sklearn`'s `OneHotEncoder `, and the default for `handle_unknown` is `error`. Therefore, if an unknown categorical feature is present during `transform`, it would through an error. 

Can you give me some code I can reproduce?

**Comment by jrzaurin** [comment]
Hey @TtCWH 

If you do not mind I am going to close this. 

if it becomes active again I will re-open :)

**Comment by chenvega** [comment]
Instead of using "preprocess_wide.fit_transform(df)", you can divide into two command:
1. preprocess_wide.fit(df)
2. X_wide_train = preprocess_wide.transform(df_train)

same as DeepPreprocessor
